app/services/participants/withdraw/validate_and_change_state.rb

```diff
@@ -9,8 +9,7 @@ module ValidateAndChangeState
       included do
         attr_accessor :reason
 
-        validates :reason, presence: true
-        validates :reason, inclusion: { in: reasons }, allow_blank: true
+        validates :reason, inclusion: { in: reasons }
```

Comment:
Why are you removing the explicit presence check? The error messages should be different, I think.